### PR TITLE
internal: splitting the 3.0 Beta and 3.0 GA feature-flags

### PR DESCRIPTION
**Pull request by tombuildsstuff**
This PR splits the 3.0 Beta and 3.0 GA feature-flags to allow the Beta functionality to be tested in the future - any one-way upgrades (such as in Storage) are left for the GA release.

### CODE REVIEW

File: internal/features/three_point_oh.go [hunk 1]
```diff
@@ -21,20 +21,30 @@ func DeprecatedInThreePointOh(deprecationMessage string) string {
...
+// ThreePointOhBeta returns whether this provider is running in 3.0 mode
+// which is an opt-in Beta of the (non-breaking changes) coming in 3.0.
+//
+// Any features behind this flag should be backwards-compatible to allow
+// users to try out 3.0 functionality.
+//
+// Whilst this currently isn't exposed, this will be via an Environment
+// Variable in the future.
+func ThreePointOhBeta() bool {
+	return false
```
Comment (manicminer) [inline internal/features/three_point_oh.go:41]: Maybe for safety?

```suggestion
	return ThreePointOh() || false
```

Comment (manicminer) [inline internal/features/three_point_oh.go:41]: Maybe for safety?

```suggestion
	return ThreePointOh() || false
```

**Review by manicminer — APPROVED** [review]
LGTM! Just a minor concern about accidentally ending up in a 'mixed mode' situation

**Review by stephybun — APPROVED** [review]

**Review by jackofallops — APPROVED** [review]
LGTM 👍

**Review by koikonom — APPROVED** [review]
:shipit:

File: internal/features/three_point_oh.go (hunk 1 above)
Comment (tombuildsstuff) [reply internal/features/three_point_oh.go:41]: I was holding off on that until the Beta flag went live, but I guess there's no harm this going in now 👍

**Pull request merged**